testing/tests/DevExpress.exporter/jspdfParts/v3/jspdf_v3.dataGrid.horizontalAlign.tests.js

```diff
@@ -792,6 +792,1195 @@ const JSPdfHorizontalAlignTests = {
             });
 
         });
+
+        QUnit.module('Pass horizontal alignment settings from dxDataGrid', moduleConfig, () => {
+
+            QUnit.test('1 col - 3 row. col.dataType: number, Cell width = 200px ', function(assert) {
+                const done = assert.async();
+                const doc = createMockPdfDoc();
+                doc.__logOptions.textOptions.hAlign = true;
+
+                const dataGrid = createDataGrid({
+                    columns: [ { dataField: 'f1', dataType: 'number', caption: 'F1' } ],
+                    dataSource: [{ f1: 1 }, { f1: 2 }],
+                });
+
+                const expectedLog = [
+                    'text,F1,210,24.2,{baseline:middle,align:right}',
+                    'text,1,210,42.6,{baseline:middle,align:right}',
+                    'text,2,210,61,{baseline:middle,align:right}',
+                    'setLineWidth,1',
+                    'rect,10,15,200,18.4',
+                    'setLineWidth,1',
+                    'rect,10,33.4,200,18.4',
+                    'setLineWidth,1',
+                    'rect,10,51.8,200,18.4'
+                ];
+
+                exportDataGrid(doc, dataGrid, { topLeft: { x: 10, y: 15 }, columnWidths: [ 200 ] }).then(() => {
+                    // doc.save(assert.test.testName + '.pdf');
+                    assert.deepEqual(doc.__log, expectedLog);
+                    done();
+                });
+            });
+
+            QUnit.test('1 col - 3 row. col.dataType: number, col.alignment: left, Cell width = 200px ', function(assert) {
+                const done = assert.async();
+                const doc = createMockPdfDoc();
+                doc.__logOptions.textOptions.hAlign = true;
+
+                const dataGrid = createDataGrid({
+                    columns: [ { dataField: 'f1', dataType: 'number', alignment: 'left', caption: 'F1' } ],
+                    dataSource: [{ f1: 1 }, { f1: 2 }],
+                });
+
+                const expectedLog = [
+                    'text,F1,10,24.2,{baseline:middle,align:left}',
+                    'text,1,10,42.6,{baseline:middle,align:left}',
+                    'text,2,10,61,{baseline:middle,align:left}',
+                    'setLineWidth,1',
+                    'rect,10,15,200,18.4',
+                    'setLineWidth,1',
+                    'rect,10,33.4,200,18.4',
+                    'setLineWidth,1',
+                    'rect,10,51.8,200,18.4'
+                ];
+
+                exportDataGrid(doc, dataGrid, { topLeft: { x: 10, y: 15 }, columnWidths: [ 200 ] }).then(() => {
+                    // doc.save(assert.test.testName + '.pdf');
+                    assert.deepEqual(doc.__log, expectedLog);
+                    done();
+                });
+            });
+
+            QUnit.test('1 col - 3 row. col.dataType: number, col.alignment: left, pdfCell.alignment: left, Cell width = 200px ', function(assert) {
```

Comment:
Many tests for [type, col.alignment, pdfCell.alignment] combinations looks redundant
imho it's enough {string, number} x {col.default , col.center} x {pdfCell. right}

Reply:
Ok, I reduced the combination of tests. At the moment there are the main scenarios affecting the redefinition of alignment value.